Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp

```diff
@@ -170,24 +170,33 @@ namespace Terrain
 
     void TerrainPhysicsColliderComponent::OnTerrainDataCreateEnd()
     {
+        m_terrainDataDestroyed = false;
```

Comment:
Suggestion: I think this would read better if you flip the logic and call it "m_terrainDataActive".